### ISSUE TITLE
debug_repl: Error when trying to parse datetime string

### ISSUE THREAD

**Issue by sysid** [zeroSteiner/rule-engine#66]
Thank you for your OSS contribution!

I am not sure what I am doing wrong when using the debug_repl. I want to parse a datetime string:
```bash
± |feat/fbs-217/bre ✗| → python -m rule_engine.debug_repl
rule > $parse_datetime("2020-01-01")
Traceback (most recent call last):
  File "/Users/Q187392/dev/z/fbs-csf/.venv/lib/python3.11/site-packages/rule_engine/debug_repl.py", line 118, in main
    result = rule.evaluate(thing)
             ^^^^^^^^^^^^^^^^^^^^
  File "/Users/Q187392/dev/z/fbs-csf/.venv/lib/python3.11/site-packages/rule_engine/engine.py", line 546, in evaluate
    return self.statement.evaluate(thing)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Q187392/dev/z/fbs-csf/.venv/lib/python3.11/site-packages/rule_engine/ast.py", line 1111, in evaluate
    return self.expression.evaluate(thing)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Q187392/dev/z/fbs-csf/.venv/lib/python3.11/site-packages/rule_engine/ast.py", line 1052, in evaluate
    function_name = function.__name__ + '?'
                    ^^^^^^^^^^^^^^^^^
AttributeError: 'functools.partial' object has no attribute '__name__'
rule >
```
Also tried: `2020-01-01T00:00:00.000000+00:00, 2020-01-01T00:00:00+00:00`.

I understand this should work. What am I doing wrong?

**Comment by zeroSteiner** [comment]
I don't think you're doing anything wrong. I'm pretty sure that's a bug.

**Comment by zeroSteiner** [comment]
Thanks for reporting this, I was able to get it fixed. The patch will be included in the next release which I'll post in the next few days.

**Comment by zeroSteiner** [comment]
This issue was fixed in [version 4.0.1](https://github.com/zeroSteiner/rule-engine/releases/tag/v4.0.1) which is now available on [PyPi](https://pypi.org/project/rule-engine/4.0.1/). Thank you for bringing it to my attention.
